plugins/modules/slack.py

```diff
@@ -143,6 +144,38 @@
       - 'never'
       - 'auto'
     version_added: 6.1.0
+  upload_file:
+    type: dict
+    description:
+      - Specify details to upload a file to Slack. The file can include metadata such as an initial comment, alt text, snipped and title.
+      - See Slack's file upload API for details at U(https://api.slack.com/methods/files.getUploadURLExternal).
+      - See Slack's file upload API for details at U(https://api.slack.com/methods/files.completeUploadExternal).
+    suboptions:
+      path:
+        type: str
+        description:
+          - Path to the file on the local system to upload.
+        required: true
+      initial_comment:
+        type: str
+        description:
+          - Optional comment to include when uploading the file.
+      alt_text:
+        type: str
+        description:
+          - Optional alternative text to describe the file.
+      snippet_type:
+        type: str
+        description:
+          - Optional snippet type for the file.
+      title:
+        type: str
+        description:
+          - Optional title for the uploaded file.
+      thread_ts:
+        type: str
+        description:
+          - Optional timestamp of parent message to thread this message, see U(https://api.slack.com/docs/message-threading).
```

Comment:
If it makes the cut today, else we'll have to update the version number
```suggestion
          - Optional timestamp of parent message to thread this message, see U(https://api.slack.com/docs/message-threading).
    version_added: 10.2.0
```